andela/ah-the-jedi-backend
# #165305264 Enable social login via Facebook, Twitter and Google.

#### What does this PR do?
Enable social login via Facebook, Twitter, and Google.
#### Description of Task to be completed?
- Implement Facebook's, Twitter's, and Google's API to allow a user to log in using either of their existing accounts
- Give the logged in user our application's JWT token.
- Write tests for the implementation.
#### How should this be manually tested?
Ensure you have the below variables in your environment.
- TWITTER_KEY=<twitter_key>
- TWITTER_SECRET=<twitter_secret>
- FACEBOOK_APP_ID=<facebook_app_id>
- FACEBOOK_SECRET_KEY=<facebook_secret_key>
- GOOGLE_CLIENT_ID= <google_client_id>
- GOOGLE_CLIENT_SECRET=<google_client_secret>
1. Clone this repo.
   `git clone https://github.com/andela/ah-the-jedi-backend.git`
2. Fetch all origin to get all branches
   `git fetch origin`
3. Checkout to `ft-enable-social-login-165305264` branch.
   `git checkout ft-enable-social-login-165305264`
4. Activate `virtualenv`
    `$ virtualenv -p python .venv`
    `$ source .venv/bin/activate`
5. Install requirements
   `$ pip install -r requirements.txt`
6. Start server
   `$ python api/manage.py runserver`
7. Test the endpoint on postman in the following steps:
- `POST /api/users/social/login/`
- Test Facebook login -> Enter provider as `facebook` and Facebook's user access token.
- Test Twitter login -> Enter provider `twitter` and Twitter's user access token and access token secret.
- Test Google login -> Enter provider as `google-oauth2` and Google's user id_token
- Use the generated JWT token to authorize the user and access protected endpoints.
#### What are the relevant pivotal tracker stories?
[#165305264](https://www.pivotaltracker.com/story/show/165305264)
#### Screenshot
Facebook Login
<img width="1131" alt="Screenshot 2019-05-01 at 12 55 17" src="https://user-images.githubusercontent.com/17095461/57013884-4b4e1700-6c16-11e9-9306-6dc0fc41fcd1.png">

Twitter Login
<img width="1129" alt="Screenshot 2019-05-01 at 12 57 41" src="https://user-images.githubusercontent.com/17095461/57013889-5143f800-6c16-11e9-8969-cabba7a43f3c.png">

Google Login
<img width="1124" alt="Screenshot 2019-05-01 at 13 13 19" src="https://user-images.githubusercontent.com/17095461/57013894-55701580-6c16-11e9-9cde-a4968a3ddd95.png">

Invalid Token
<img width="1125" alt="Screenshot 2019-05-01 at 13 42 09" src="https://user-images.githubusercontent.com/17095461/57014097-2efeaa00-6c17-11e9-8b63-9bae3e06d2f5.png">
